src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,63 @@
+package duke;
+
+import java.io.IOException;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.util.Parser;
+import duke.util.Ui;
+
+/**
+ * The {@code Duke} class is the entry point of duke.
+ * To begin the program, it calls run() on an instance of this class.
+ */
+public class Duke {
+    private final TaskList tasks;
+    private final String pathName = "./data/duke.txt";
+    private final Storage storage;
+    private final Ui ui;
+
+    /**
+     * A private constructor to initialize variables.
+     *
+     * @throws IOException Invalid file.
+     * @throws DukeException Throw respective exceptions.
+     */
+    private Duke() throws IOException, DukeException {
+        storage = new Storage(pathName);
+        ui = new Ui();
+
```

Comment:
Is this extra line between lines 29 and 31 needed?

Reply:
Will remove